src/frontends/paddle/src/op/set_value.cpp

```diff
@@ -0,0 +1,120 @@
+// // Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+#define MAX_VALUE(T) std::numeric_limits<T>::max()
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+
+std::shared_ptr<default_opset::Constant> get_max_value_by_dtype(ov::element::Type dtype) {
+    if (dtype == element::f32)
+        return default_opset::Constant::create(dtype, {}, {MAX_VALUE(float)});
+    else
+        return default_opset::Constant::create(dtype, {}, {MAX_VALUE(int)});
+};
+
+Output<Node> handle_minus_index(const OutputVector& node, const Output<Node>& dim) {
+    const auto zero = default_opset::Constant::create(node[0].get_element_type(), {1}, {0});
+    const auto concat_node = std::make_shared<default_opset::Concat>(node, 0);
+    const auto mask = std::make_shared<default_opset::Less>(concat_node, zero);
+    const auto res = std::make_shared<default_opset::Add>(concat_node, dim);
+    return std::make_shared<default_opset::Select>(mask, res, concat_node);
+}
+
+NamedOutputs set_value(const NodeContext& node) {
+    const auto input_node = node.get_input("Input");
+    auto value_node = node.get_input("ValueTensor");
+    const auto input_shape = input_node.get_partial_shape().get_shape();
+    const auto dims = static_cast<int64_t>(input_node.get_partial_shape().rank().get_length());
+    const auto dtype = input_node.get_element_type();
+    const auto axes = node.get_attribute<std::vector<int64_t>>("axes");
+
+    Output<Node> padding_starts_node, padding_ends_node, value_target_shape;
+    Shape value_shape(input_shape);
+
+    if (node.has_input("StartsTensorList") && node.has_input("StepsTensorList") && node.has_input("EndsTensorList")) {
+        const auto dim_node = default_opset::Constant::create(element::i64, {input_shape.size()}, input_shape);
+        auto starts = handle_minus_index(node.get_ng_inputs("StartsTensorList"), dim_node);
```

Comment:
I think it should not be 'dim_node', it should be [input_shape[axes[i]] for i in len(starts)]

Reply:
Yes, you are right. Test cases are not comprehensive enough